### ISSUE TITLE
EmailAddress string raises exception when user does not exist

### ISSUE THREAD

**Issue by thenewguy** [pennersr/django-allauth#2373]
EmailAddress string method needs to check that user is set before attemping to force it into the string

```
 Traceback (most recent call last):
   File "/usr/lib/python2.7/logging/__init__.py", line 851, in emit
     msg = self.format(record)
   File "/usr/lib/python2.7/logging/__init__.py", line 724, in format
     return fmt.format(record)
   File "/usr/lib/python2.7/logging/__init__.py", line 464, in format
     record.message = record.getMessage()
   File "/usr/lib/python2.7/logging/__init__.py", line 328, in getMessage
     msg = msg % self.args
   File "/usr/local/lib/python2.7/dist-packages/django/db/models/base.py", line 590, in __repr__
     u = six.text_type(self)
   File "/usr/local/lib/python2.7/dist-packages/allauth/account/models.py", line 40, in __str__
     return "%s (%s)" % (self.email, self.user)
   File "/usr/local/lib/python2.7/dist-packages/django/db/models/fields/related_descriptors.py", line 194, in __get__
     "%s has no %s." % (self.field.model.__name__, self.field.name)
 RelatedObjectDoesNotExist: EmailAddress has no user.
```

**Comment by thenewguy** [comment]
Please let me know if this issue is accepted and I will create a patch and PR

**Comment by Bartleby2718** [comment]
If you can, just create a pull request yourself--mentioning this issue--and wait for the maintainer's response. Be wary that it takes very long for the maintainer to look at it. (It took me more than a month.)
FYI Python 2 won't be supported: https://github.com/pennersr/django-allauth/commit/7b81531bc89ae98dc6f687611743db5b36cda9a2

**Comment by pennersr** [comment]
Closed via f0c1defd5c4bbe33ab8eb9d5811d9b16f726b73e

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/2373
